.github/workflows/quality-gate.yml

```diff
@@ -0,0 +1,40 @@
+name: Quality Gate
+
+on: [push, pull_request, workflow_dispatch]
```

Comment:
What is `workflow_dispatch` for?

Reply:
To manually trigger the action if needed. 

https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/